conf/config.go

```diff
@@ -277,38 +288,41 @@ func Checkout(v interface{}) error {
 
 }
 
-// walk is a private function used by Checkout to recursively walk a possible nested struct.
-func walk(field reflect.Value) {
-
-	// Get the number of fields in the struct
-	num := field.NumField()
-	// Used to get Tag information
-	fieldType := field.Type()
-
-	// Traverse the struct
-	for i := 0; i < num; i++ {
-		innerField := field.Field(i)
-		// Ensure the field is "exportable" and is a type string
-		if innerField.CanInterface() && innerField.IsValid() && innerField.Type().Name() == "string" {
-
-			// Tags are supported
-			name := fieldType.Field(i).Tag.Get("conf")
-			// If "-" is provided as a tag, explicitly skip that field
-			if name == "-" {
-				continue
-			} else if name == "" {
-				// If the Tag is not there, then default to the name of the struct
-				name = fieldType.Field(i).Name
+// bindenv: workaround to make the unmarshal work with environment variables
+// Inspired from solution found here : https://github.com/spf13/viper/issues/188#issuecomment-399884438
+func bindenvs(field reflect.Value, parts ...string) error {
+	if field.Kind() == reflect.Ptr {
+		field = field.Elem()
```

Comment:
Do you have any, or is there any chance, we will have a config variable behind a pointer / reference?  If not, I wonder if we can just replace this with a continue and skip that field?